workers/generic-worker/multiuser_linux_test.go

```diff
@@ -3,47 +3,21 @@
 package main
 
 import (
-	"strings"
 	"testing"
 
 	"github.com/mcuadros/go-defaults"
 )
 
-func TestTaskUserCredentialsEnvVarIsWrittenAsCurrentUser(t *testing.T) {
-	setup(t)
-	config.EnableRunTaskAsCurrentUser = true
-
-	payload := GenericWorkerPayload{
-		Command: [][]string{{
-			"echo", "${TASK_USER_CREDENTIALS:?Error: TASK_USER_CREDENTIALS not set}",
-		}},
-		MaxRunTime: 180,
-		Features: FeatureFlags{
-			RunTaskAsCurrentUser: true,
-		},
-	}
-	defaults.SetDefaults(&payload)
-
-	td := testTask(t)
-	td.Scopes = append(td.Scopes,
-		"generic-worker:run-task-as-current-user:"+td.ProvisionerID+"/"+td.WorkerType,
-	)
-
-	_ = submitAndAssert(t, td, payload, "completed", "completed")
-	logtext := LogText(t)
-	if strings.Contains(logtext, "Error: TASK_USER_CREDENTIALS not set") {
-		t.Fatalf("Was expecting log file to not contain \"Error: TASK_USER_CREDENTIALS not set\", but it does: \n%v", logtext)
-	}
-}
-
 func TestXDGRuntimeDirEnvVarIsRemovedAsCurrentUser(t *testing.T) {
 	setup(t)
 	config.EnableRunTaskAsCurrentUser = true
 
 	payload := GenericWorkerPayload{
-		Command: [][]string{{
-			"echo", "${XDG_RUNTIME_DIR:?Success: XDG_RUNTIME_DIR not set}",
-		}},
+		Command: goRun(
+			"check-env.go",
+			"XDG_RUNTIME_DIR",
+			"/run/user/1000",
+		),
 		Env: map[string]string{
 			"XDG_RUNTIME_DIR": "/run/user/1000",
 		},
```

Comment:
i don't think you want to specify it in the environment here, because the feature likely overwrites it with a different value (i.e. not /run/user/1000) and then it will not match, and the task will fail, so the test will pass. I think in this case, your previous implementation might have been better, if it was working. Just because we don't know what the value will be...

Reply:
The feature _should_ entirely remove this variable though.